public/js/audioplayer.js

```diff
@@ -0,0 +1,209 @@
+/*
+    By Osvaldas Valutis, www.osvaldas.info
+    Available for use under the MIT License
+*/
+
+
+
+;(function( $, window, document, undefined )
+{
+    var isTouch       = 'ontouchstart' in window,
+        eStart        = isTouch ? 'touchstart'  : 'mousedown',
+        eMove         = isTouch ? 'touchmove'   : 'mousemove',
+        eEnd          = isTouch ? 'touchend'    : 'mouseup',
+        eCancel       = isTouch ? 'touchcancel' : 'mouseup',
+        secondsToTime = function( secs )
+        {
+            var hoursDiv = secs / 3600, hours = Math.floor( hoursDiv ), minutesDiv = secs % 3600 / 60, minutes = Math.floor( minutesDiv ), seconds = Math.ceil( secs % 3600 % 60 );
+            if( seconds > 59 ) { seconds = 0; minutes = Math.ceil( minutesDiv ); }
+            if( minutes > 59 ) { minutes = 0; hours = Math.ceil( hoursDiv ); }
+            return ( hours == 0 ? '' : hours > 0 && hours.toString().length < 2 ? '0'+hours+':' : hours+':' ) + ( minutes.toString().length < 2 ? '0'+minutes : minutes ) + ':' + ( seconds.toString().length < 2 ? '0'+seconds : seconds );
+        },
+        canPlayType   = function( file )
+        {
+            var audioElement = document.createElement( 'audio' );
+            return !!( audioElement.canPlayType && audioElement.canPlayType( 'audio/' + file.split( '.' ).pop().toLowerCase() + ';' ).replace( /no/, '' ) );
+        };
+
+    $.fn.audioPlayer = function( params )
+    {
+        var params      = $.extend( { classPrefix: 'audioplayer', strPlay: 'Play', strPause: 'Pause', strVolume: 'Volume' }, params ),
+            cssClass    = {},
+            cssClassSub =
+            {
+                playPause:      'playpause',
+                playing:        'playing',
+                stopped:        'stopped',
+                time:           'time',
+                timeCurrent:    'time-current',
+                timeDuration:   'time-duration',
+                bar:            'bar',
+                barLoaded:      'bar-loaded',
+                barPlayed:      'bar-played',
+                volume:         'volume',
+                volumeButton:   'volume-button',
+                volumeAdjust:   'volume-adjust',
+                noVolume:       'novolume',
+                muted:          'muted',
+                mini:           'mini'
+            };
+
+        for( var subName in cssClassSub )
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.